pkg/controller/storagecluster/reconcile.go

```diff
@@ -626,23 +666,32 @@ func (r *ReconcileStorageCluster) ensureCephConfig(sc *ocsv1.StorageCluster, req
 // ensureCephCluster ensures that a CephCluster resource exists with its Spec in
 // the desired state.
 func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
-	// if StorageClass is "gp2" or "io1" based, set tuneSlowDeviceClass to true
-	// this is for performance optimization of slow device class
-	//TODO: If for a StorageDeviceSet there is a separate metadata pvc template, check for StorageClass of data pvc template only
-	for i, ds := range sc.Spec.StorageDeviceSets {
-		throttle, err := r.throttleStorageDevices(*ds.DataPVCTemplate.Spec.StorageClassName)
-		if err != nil {
-			return fmt.Errorf("Failed to verify StorageClass provisioner. %+v", err)
-		}
-		if throttle {
-			sc.Spec.StorageDeviceSets[i].Config.TuneSlowDeviceClass = true
-		} else {
-			sc.Spec.StorageDeviceSets[i].Config.TuneSlowDeviceClass = false
+	if !sc.Spec.ExternalStorage.Enable {
```

Comment:
Actually, for an external CephCluster I think we shouldn't be doing anything with StorageDeviceSets at all! What would you think about instead putting in an `if sc.Spec.ExternalStorage.Enable && sc.Spec.StorageDeviceSets != nil` clause that would throw an error? That way we require that there be no StorageDevices defined.

Reply:
Got your point... 

I have made the change as you suggested, but has put `len(sc.Spec.StorageDeviceSets) != 0` approach and the logic behind was `len(<array>)` function will return `zero` for both nil and non-nil empty array.